### PR TITLE
Configure specific File and Folder Permissions

### PR DESCRIPTION
**Pull request by abraunegg**
* Initial work on custom file and folder permissions

### CODE REVIEW

**Comment by abraunegg** [comment]
@norbusan 
This PR sets the file | folder defaults to:
```
        // Default directory permission mode
	public long defaultDirectoryPermissionMode = 755;
	// Default file permission mode
	public long defaultFilePermissionMode = 644;
```

What are your thoughts here?

**Comment by norbusan** [comment]
I would probably go for 700/600 as default when creating the first directory, and then use the permissions of the parent folder as default.

File: src/main.d [hunk 1]
```diff
@@ -607,20 +607,41 @@ int main(string[] args)
 		itemDb = new ItemDatabase(cfg.databaseFilePathDryRun);
 	}
 	
+	// What are the permission that have been set for the application?
+	// These are relevant for:
+	// - The ~/OneDrive parent folder or 'sync_dir' configured item
+	// - Any new folder created under ~/OneDrive or 'sync_dir'
+	// - Any new file created under ~/OneDrive or 'sync_dir'
+	// valid permissions are 000 -> 777 - anything else is invalid
+	if ((cfg.getValueLong("sync_dir_permissions") > 777) || (cfg.getValueLong("sync_file_permissions") > 777)) {
```
Comment (norbusan) [inline src/main.d:616]: I think we allow values <0 here. The ValueLong uses `to!long` for conversion and (but I am not sure!) I guess that this accepts negative numbers, too?

Comment (abraunegg) [reply src/main.d:616]: Yes .. good catch .. need to trap for negative as well

**Comment by abraunegg** [comment]
> I would probably go for 700/600 as default when creating the first directory, and then use the permissions of the parent folder as default.

Done - defaults now 700|600

**Review by norbusan — APPROVED** [review]

**Pull request merged**